File: language-wars/csv-to-xml/ruby/csv-to-xml.rb
puts "<records>"
  STDIN.read.split("\n").each do |r| 
    puts "  <record>"
    r.split(",").each do |f|
      puts " " * 4 + "<field>#{f}</field>"
    end
    puts "  </record>"
  end
puts "</records>"

File: language-wars/csv-to-xml/scala/csv-to-xml.scala
println(
  new xml.PrettyPrinter(80, 2).format(
    <records>{
      io.Source.stdin.getLines.map { ln =>
        <record>{ln.split(",").map {f => <field>{f}</field>}}</record>
      }
    }</records>
  )
)
